ImageInLib/ImageInLib/src/segmentation2D_lagrangean.c

```diff
@@ -75,25 +73,31 @@ bool lagrangeanExplicit2DCurveSegmentation(Image_Data2D inputImage2D, const Lagr
     }
 
     //get edge detector
-    
     for (size_t i = 0; i < dataDimension; i++)
     {
-        edge_detector[i] = edgeDetector(abs_val_grad[i], edge_detector_coef);
+        edge_detector[i] = edgeDetector(abs_val_grad[i], pSegmentationParams->edgeCoef);
     }
 
+    unsigned char saveEdge[] = "C:/Users/Konan Allaly/Documents/Tests/Curves/Output/edge_detector.raw";
+    manageFile(edge_detector, 0, inputImage2D.width, inputImage2D.height, saveEdge, STORE_2D_RAW_DATA, BINARY_DATA, (Storage_Flags) { false, false });
+
     Point2D centroid = getCurveCentroid(pSegmentationParams->pinitial_condition);
     size_t centroid_i = (size_t)(centroid.y + 0.5);
     size_t centroid_j = (size_t)(centroid.x + 0.5);
-    dataType ref_intensity = inputImage2D.imageDataPtr[x_new(centroid_j, centroid_i, inputImage2D.width)];
+    //The reference intensity is now computed in the main and incorporated to the segmentation parameters
+    //dataType ref_intensity = inputImage2D.imageDataPtr[x_new(centroid_j, centroid_i, inputImage2D.width)];
 
     if (!pSegmentationParams->open_curve)
     {
         for (size_t i = 0; i < dataDimension; i++)
         {
-            similar_intensity_detector[i] = similarIntensityDetector(inputImage2D.imageDataPtr[i], ref_intensity, similar_intensity_detector_coef);
+            similar_intensity_detector[i] = similarIntensityDetector(inputImage2D.imageDataPtr[i], pSegmentationParams->reference_intensity, pSegmentationParams->intensityCoef);
         }
     }
 
+    unsigned char saveG2[] = "C:/Users/Konan Allaly/Documents/Tests/Curves/Output/similarIntensityDetect.raw";
```

Comment:
please remove or at least comment out the absolute path

Reply:
done